src/rgw/rgw_sal_motr.cc

```diff
@@ -1923,127 +1922,130 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
       if(ent.key.instance == "null")
         del_null_ref_key = true;
     }
```

Comment:
We need to delete the null reference entry once we remove the delete marker. Also, in case of unversioned and suspended scenarios, this check is required.

Reply:
We should not touch previous null object while adding or deleting a deleteMarker.

Reply:
In case of suspended bucket scenarios, at a time there will be only one null entry, it can be null version or null delete marker. 
- When we delete an object in suspended bucket, we add a null delete marker and add it's reference to null entry(obj/null). Also, we delete the previous null version object
- Hence, we require to delete that entry when we delete the delete marker.

This code requires some optimization which we will cover as a part of another JIRA - https://jts.seagate.com/browse/CORTX-31978